corehq/apps/hqwebapp/static/hqwebapp/js/downgrade_modal.js

```diff
@@ -0,0 +1,49 @@
+/*
+    To use, include this file on a page that also includes hqwebapp/downgrade_modal.html
+*/
+hqDefine("hqwebapp/js/downgrade_modal", [
```

Comment:
This and `rollout_modal.js` look about 75% the same - how deeply did you go into sharing code between them?

Reply:
I thought about this because the structures were so similar but couldn't think of good ways to do this that didn't involve introducing some inversion of control, which feels more complex than just having a separate ```downgrade_modal.js```.  But if you have ideas I'd like to hear them.

Reply:
I had imagined having one js file and two different modals, but looking closer, that wouldn't account for the logic that the rollout modal needs to run to enable the feature. I think it would be fairly straightforward if these were knockout controlled and you had one base SnoozeModal model for this functionality and then a RolloutModal that inherits from that one and adds in the enable/disable behavior. But that's probably out of scope.